### PR TITLE
feat: using connect function and building sdk instance out of provider

### PR DESCRIPTION
**Pull request by victormer**

### CODE REVIEW

**Review by AbdullahWins — COMMENTED** [review]
Well Done!

File: state/human.context.js [hunk 1]
```diff
@@ -19,12 +19,21 @@ const HumanContext = createContext({
 const uri = process.env.NEXT_PUBLIC_GRAPHQL_ENDPOINT
 const projectId = process.env.NEXT_PUBLIC_PROJECT_ID
 
+// We build the HumanWalletSDK instance here
```
Comment (danivm) [inline state/human.context.js:22]: 👌

File: state/human.context.js [hunk 2]
```diff
@@ -57,6 +69,7 @@ function HumanProvider(props) {
   }
 
   const confirmProposal = async ({ proposalId, code }) => {
+    if (!humanSDK.isReady()) return
```
Comment (davecarter) [inline state/human.context.js:72]: this guard is [already in the SDK](https://github.com/Tutellus/humanWalletSDK/blob/develop/src/index.js#L160), so you don't have to care about here

File: state/human.context.js [hunk 3]
```diff
@@ -70,7 +83,7 @@ function HumanProvider(props) {
   }
 
   const getTokensBalance = async () => {
-    if (!human) return
+    if (!humanSDK.isReady()) return
```
Comment (davecarter) [inline state/human.context.js:86]: same here
the thing is to be sure the async state of the client (PoC) has been properly updated

File: state/human.context.js [hunk 4]
```diff
@@ -19,12 +19,21 @@ const HumanContext = createContext({
 const uri = process.env.NEXT_PUBLIC_GRAPHQL_ENDPOINT
 const projectId = process.env.NEXT_PUBLIC_PROJECT_ID
 
+// We build the HumanWalletSDK instance here
+const humanSDK = HumanWalletSDK.build({
```
Comment (davecarter) [inline state/human.context.js:23]: that's cool, let's see

File: state/human.context.js [hunk 5]
```diff
@@ -19,12 +19,21 @@ const HumanContext = createContext({
 const uri = process.env.NEXT_PUBLIC_GRAPHQL_ENDPOINT
```
Comment (davecarter) [inline state/human.context.js:19]: missing in vercel

**Review by davecarter — APPROVED** [review]

File: state/human.context.js (hunk 5 above)
Comment (victormer) [reply state/human.context.js:19]: solved

**Pull request merged**